pkg/controller/syncconfig/utils_test.go

```diff
@@ -0,0 +1,192 @@
+package syncconfig
+
+import (
+	ctx "context"
+	"testing"
+
+	"github.com/vshn/espejo/pkg/apis/sync/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func TestProcessTemplate(t *testing.T) {
+	obj := &unstructured.Unstructured{}
+	obj.SetAnnotations(map[string]string{"replace": "${PROJECT_NAME}"})
+	obj.SetNamespace("name-with-suffix-${PROJECT_NAME}")
+
+	processTemplate(obj, "test-name")
+
+	if obj.GetNamespace() != "name-with-suffix-test-name" {
```

Comment:
Using `github.com/stretchr/testify/assert` for test assertions makes writing and reading tests much easier.